src/Timezone/UserBasedTimezoneDetector.php

```diff
@@ -24,20 +23,10 @@
  */
 class UserBasedTimezoneDetector implements TimezoneDetectorInterface
 {
-    /**
-     * @var SecurityContextInterface|TokenStorageInterface
-     */
     protected $securityContext;
 
-    /**
-     * @param SecurityContextInterface|TokenStorageInterface $securityContext
-     */
-    public function __construct($securityContext)
+    public function __construct(TokenStorageInterface $securityContext)
```

Comment:
Shouldn't we move "symfony/security-core" from "require-dev" to "require"?

Reply:
This was introduced in `require-dev` in https://github.com/sonata-project/SonataIntlBundle/commit/fd3f09fb0d09296e5968a030df0a8eedcd6b3a0d , because it was optional at the time. In that same commit, the service definition is removed if the user bundle is not registered. If that's still the case, we should leave that here.

Reply:
I understand, but from a strict perspective (regardless a service using this class is present or not), it's available when the package is installed.
I don't know if we can mark this as internal.

Reply:
We have quite a few packages having this concept of optional dependencies, and usually use `require-dev` for those, so if we are changing policies, a lot of things will probably need to change. See for example the media bundle.